confluent/confluent.go

```diff
@@ -7,30 +7,62 @@ import (
 	"fmt"
 	"github.com/confluentinc/confluent-kafka-go/kafka"
 	"github.com/magiconair/properties"
+	k "github.com/osframework/confluent-kafka-go-ext/kafka"
 	log "github.com/sirupsen/logrus"
+	"strconv"
 	"time"
 )
 
+const (
+	AdminOperationTimeout = "admin.operation.timeout"
+	AutoOffsetReset       = "auto.offset.reset"
+	GroupId               = "group.id"
+	NumPartitions         = "topic.partitions"
+	ReplicationFactor     = "topic.replication.factor"
+
+	DefaultAdminOperationTimeout = "60s"
+	DefaultNumPartitions         = "1"
+	DefaultReplicationFactor     = "3"
+)
+
+type NewAdminClient func(p k.Producer) (k.AdminClient, error)
+
 // Create a new Kafka consumer, using the specified configuration settings. An
 // error will be returned if the given configuration does not provide a consumer
 // group ID, or if the consumer cannot connect to the specified host for any
 // reason.
-func NewConsumer(config map[string]string) (*kafka.Consumer, error) {
+func NewConsumer(config map[string]string) (k.Consumer, error) {
 	configMap := createBasicConfigMap(config)
-	err := configMap.SetKey("group.id", config["group.id"])
-	err = configMap.SetKey("auto.offset.reset", config["auto.offset.reset"])
+	settingsToValidate := [2]string{GroupId, AutoOffsetReset}
+	for _, setting := range settingsToValidate {
+		v, err := configMap.Get(setting, "")
+		if nil != err {
```

Comment:
I've never seen this written this way. I assume it still works, but it looks very odd

Reply:
For many years now, on equality and identity checks I put the constant value on the left side vs the right side. It eliminates a classic source of null pointer errors, yet it is computationally equivalent to the traditional style that places the variable on the left.